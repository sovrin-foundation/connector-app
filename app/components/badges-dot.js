// @flow
import React, { Component, PureComponent } from 'react'
import { View, StyleSheet, Platform } from 'react-native'
import { HAIRLINE_WIDTH, maroonRed, white } from '../common/styles/constant'

type DotProps = {
  size: string,
}

export class Dot extends PureComponent<DotProps, void> {
  render() {
    const { size } = this.props
    const dotSize =
      size === 'small'
        ? styles.badgeSmall
        : size === 'medium' ? styles.badgeMedium : styles.badgeLarge
    return <View style={[styles.badge, dotSize]} />
  }
}

const styles = StyleSheet.create({
  badge: {
    backgroundColor: maroonRed,
    ...Platform.select({
      ios: {
        borderWidth: HAIRLINE_WIDTH,
      },
      android: {
        borderWidth: 1,
      },
    }),
    borderColor: white,
  },
  badgeSmall: {
    width: 15,
    height: 15,
    borderRadius: 10,
  },
  badgeMedium: {
    width: 20,
    height: 20,
    borderRadius: 10,
  },
  badgeLarge: {
    width: 25,
    height: 25,
    borderRadius: 15,
  },
})
