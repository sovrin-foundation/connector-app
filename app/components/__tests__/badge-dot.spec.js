// @flow
import React from 'react'
import 'react-native'
import { Dot } from '../badges-dot'
import renderer from 'react-test-renderer'

describe('<Dot />', () => {
  it('should render with small Dot', () => {
    const dotComponent = renderer.create(<Dot size={'small'} />).toJSON()
    expect(dotComponent).toMatchSnapshot()
  })
  it('should render with medium Dot', () => {
    const dotComponent = renderer.create(<Dot size={'medium'} />).toJSON()
    expect(dotComponent).toMatchSnapshot()
  })
  it('should render with large Dot', () => {
    const dotComponent = renderer.create(<Dot size={'large'} />).toJSON()
    expect(dotComponent).toMatchSnapshot()
  })
})
