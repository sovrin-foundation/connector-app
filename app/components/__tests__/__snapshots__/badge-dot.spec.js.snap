// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dot /> should render with large Dot 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#d1021b",
        "borderColor": "#ffffff",
        "borderWidth": 0.2,
      },
      Object {
        "borderRadius": 15,
        "height": 25,
        "width": 25,
      },
    ]
  }
/>
`;

exports[`<Dot /> should render with medium Dot 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#d1021b",
        "borderColor": "#ffffff",
        "borderWidth": 0.2,
      },
      Object {
        "borderRadius": 10,
        "height": 20,
        "width": 20,
      },
    ]
  }
/>
`;

exports[`<Dot /> should render with small Dot 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#d1021b",
        "borderColor": "#ffffff",
        "borderWidth": 0.2,
      },
      Object {
        "borderRadius": 10,
        "height": 15,
        "width": 15,
      },
    ]
  }
/>
`;
